client/components/tinymce/plugins/embed/dialog.jsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Dialog from 'components/dialog';
+import FormTextInput from 'components/forms/form-text-input';
+import { localize } from 'i18n-calypso';
+
+/*
+ * Shows the URL of am embed and allows it to be edited.
+ */
+export class EmbedDialog extends React.Component {
+	static propTypes = {
+		embedUrl: PropTypes.string,
+		isVisible: PropTypes.bool,
+
+		// Event handlers
+		onCancel: PropTypes.func.isRequired,
+		onUpdate: PropTypes.func.isRequired,
+
+		// Inherited
+		translate: PropTypes.func.isRequired,
+	};
+
+	static defaultProps = {
+		embedUrl: '',
+		isVisible: false,
+	};
+
+	state = {
+		embedUrl: this.props.embedUrl,
+	};
+
+	/**
+	 * Reset `state.embedUrl` whenever the component's dialog is opened or closed.
+	 *
+	 * If this were not done, then switching back and forth between multiple embeds would result in
+	 * `state.embedUrl` being incorrect. For example, when the second embed was opened,
+	 * `state.embedUrl` would equal the value of the first embed, since it initially set the
+	 * state.
+	 *
+	 * @param {object} nextProps The properties that will be received.
+	 */
+	componentWillReceiveProps = ( nextProps ) => {
+		this.setState( {
+			embedUrl: nextProps.embedUrl,
+		} );
+	};
+
+	onChangeEmbedUrl = ( event ) => {
+		this.setState( {
+			embedUrl: event.target.value,
+		} );
+	};
+
+	onUpdate = () => {
+		this.props.onUpdate( this.state.embedUrl );
+	};
+
+	onKeyDownEmbedUrl = ( event ) => {
+		if ( 'Enter' !== event.key ) {
+			return;
+		}
+
+		event.preventDefault();
+		this.onUpdate();
+	};
+
+	render() {
+		const { translate } = this.props,
+			dialogButtons = [
```

Comment:
I'm not sure it's in the guidelines but the trend is to define each `const` or `let` separately - I believe for the same reasons mentioned in [AirBnBs guidelines](https://github.com/airbnb/javascript#variables--one-const).

Reply:
Ah, nice! I often prefer that as well. It doesn't look like there's anything about it in the docs, I think I was just doing it out of habit from the Core guidelines.